eventuals/task.h

```diff
@@ -238,41 +282,71 @@ struct _TaskFromToWith {
     K_ k_;
     std::tuple<Args_...> args_;
 
-    Callback<
-        Action,
-        std::optional<std::exception_ptr>&&,
-        Args_&&...,
-        std::optional<
-            std::conditional_t<
-                std::is_void_v<From_>,
-                std::monostate,
-                From_>>&&,
-        std::unique_ptr<void, Callback<void*>>&,
-        Interrupt&,
+    // The 'dispatch_' is a `std::variant` because its either a function that
+    // creates an eventual or the value from 'Task::Success' that should be
+    // passed on to the continuation.
+    std::variant<
         std::conditional_t<
             std::is_void_v<To_>,
-            Callback<>&&,
-            Callback<To_>&&>,
-        Callback<std::exception_ptr>&&,
-        Callback<>&&>
+            std::monostate,
+            std::conditional_t<
+                !std::is_reference_v<To_>,
+                To_,
+                std::reference_wrapper<
+                    std::remove_reference_t<To_>>>>,
+        Dispatch_>
         dispatch_;
 
     std::unique_ptr<void, Callback<void*>> e_;
     Interrupt* interrupt_ = nullptr;
   };
 
-  template <typename From_, typename To_, typename... Args_>
+  template <
+      typename From_,
+      typename To_,
+      typename... Args_>
   struct Composable {
+    using Dispatch_ =
```

Comment:
Let's actually move this into `struct _TaskFromToWith` and do:

```cpp
template <typename From, typename To>
using DispatchCallback = Callback<...>;
```

Then we can use `DispatchCallback` in both `struct Composable` and `struct Continuation`.

Reply:
There should be `Args` also.